exporters/jaeger/src/THttpTransport.h

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <opentelemetry/exporters/jaeger/jaeger_http.h>
+#include <opentelemetry/ext/http/client/http_client.h>
+#include <opentelemetry/version.h>
+
+#include <thrift/transport/TVirtualTransport.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace jaeger
+{
+
+class THttpTransport : public apache::thrift::transport::TVirtualTransport<THttpTransport>
+{
+public:
+  THttpTransport(std::string endpoint, const std::vector<HttpHeader> &extra_headers);
+  ~THttpTransport() override;
+
+  bool isOpen() const override;
+
+  uint32_t read(uint8_t *buf, uint32_t len);
```

Comment:
Do we need to provide `read` as it is not used?

Reply:
The base Transport class (`TTransport::read_virt`) will throw if this is invoked and I wasn't sure if the Thrift binary protocol will actually call this method, probably not, but just to be on the safe side :fearful: